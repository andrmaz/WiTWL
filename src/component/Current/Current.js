import React from 'react';
import PropTypes from 'prop-types';
import './Current.scss';

const Current = ({ weather }) => {
    const { current } = weather;
    const text = current?.condition.text ? current.condition.text : '';
    const update = current?.last_updated ? current.last_updated : '';
    return (
        <section className='box3'>
            <article className="wrapper">
                <h1>Current weather</h1>
                <p>{`Condition: ${text}`}</p>
                {
                    current?.condition.icon ?
                    <img 
                        src={`http:${current?.condition.icon}`} 
                        alt='weather' 
                    /> : ''
                }
                <p>{`Last updated: ${update}`}</p>
            </article>
        </section>
    );
};

Current.propTypes = {
    weather: PropTypes.object
};

export default Current;
