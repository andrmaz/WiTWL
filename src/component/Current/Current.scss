.box3 {
    grid-row-start: 1;
    grid-row-end: 3;
    grid-column-start: 2;
    grid-column-end: 3;

    .wrapper {
        height: 80%;
        width: 70%;
        display: flex;
        flex-direction: column;
        justify-content: space-evenly;
        align-items: center;
    }
}

@media screen and (max-width: 1024px) {
    .box3 {
        grid-row-start: 2;
        grid-row-end: 3;
        grid-column-start: 2;
        grid-column-end: 3;

        .wrapper {
            height: 100%;
            width: 100%;
        }
    }
}