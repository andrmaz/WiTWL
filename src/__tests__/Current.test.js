import React from 'react';
import { shallow } from 'enzyme';
import Current from '../component/Current/Current';

let wrapper;
const mockInitialWeather = {};
const mockWeather = { 
    current: {
        condition: { text: '', icon: ''},
        last_updated: ''
    }
};

describe('Current component', () => {
    it('should render component correctly', () => {
        wrapper = shallow(<Current weather={mockWeather}/>);
        expect(wrapper).toMatchSnapshot()
    })
    it('should return empty strings on initial state', () => {
        wrapper = shallow(<Current weather={mockInitialWeather}/>);
        expect(wrapper.text()).toMatch('Condition:')
    })
})