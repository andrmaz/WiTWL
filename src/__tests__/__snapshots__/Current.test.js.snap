// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current component should render component correctly 1`] = `
<section
  className="box3"
>
  <article
    className="wrapper"
  >
    <h1>
      Current weather
    </h1>
    <p>
      Condition: 
    </p>
    <p>
      Last updated: 
    </p>
  </article>
</section>
`;
